src/Stratis.Bitcoin/P2P/Protocol/CallbackMessageListener.cs

```diff
@@ -23,15 +24,17 @@ namespace Stratis.Bitcoin.P2P.Protocol
     public class CallbackMessageListener<T> : IMessageListener<T>, IDisposable
     {
         /// <summary>Queue of the unprocessed incoming messages.</summary>
-        private readonly AsyncQueue<T> asyncQueue;
+        private readonly IAsyncDelegateDequeuer<T> asyncQueue;
+        private readonly IAsyncProvider asyncProvider;
```

Comment:
Do you need to keep this instance here?

Reply:
probably not because we use it only in constructor